examples/plot_meg_data_example.py

```diff
@@ -93,34 +92,89 @@ def _compute_stc(zscore_active_set, active_set, evoked, forward):
-cov_fname = data_path + '/MEG/sample/sample_audvis-shrunk-cov.fif'
-subjects_dir = data_path + '/subjects'
-condition = 'Left Auditory'
-
-# Read noise covariance matrix
-noise_cov = mne.read_cov(cov_fname)
-
-# Handling average file
-evoked = mne.read_evokeds(ave_fname, condition=condition, baseline=(None, 0))
-evoked = evoked.pick_types('grad')
-# Selecting relevant time window
-evoked.plot()
-evoked.crop(tmin=0.05, tmax=0.1)
-
-# Handling forward solution:
+if cond in ['audio', 'visual']:
+
+    subject = 'sample'
+    data_path = sample.data_path()
+    fwd_fname_suffix = 'MEG/sample/sample_audvis-meg-eeg-oct-6-fwd.fif'
+    fwd_fname = os.path.join(data_path, fwd_fname_suffix)
+    ave_fname = os.path.join(data_path, 'MEG/sample/sample_audvis-ave.fif')
+    raw_fname = os.path.join(data_path, 'MEG/sample/sample_audvis_raw.fif')
+    cov_fname_suffix = 'MEG/sample/sample_audvis-shrunk-cov.fif'
+    cov_fname = os.path.join(data_path, cov_fname_suffix)
+    cov_fname = data_path + '/' + cov_fname_suffix
+    subjects_dir = os.path.join(data_path, 'subjects')
+
+    if cond == 'audio':
+        condition = 'Left Auditory'
+    elif cond == 'visual':
+        condition = 'Left visual'
+
+    # Read noise covariance matrix
+    noise_cov = mne.read_cov(cov_fname)
+
+    # Handling average file
+    evoked = mne.read_evokeds(ave_fname, condition=condition,
+                              baseline=(None, 0))
+    evoked = evoked.pick_types('grad')
+
+    # Selecting relevant time window
+    evoked.plot()
+    t_min, t_max = 0.05, 0.1
+    t_step = 0.01
+
+    pca = False
+
+elif cond == 'somato':
+
+    data_path = somato.data_path()
+    subject = '01'
+    subjects_dir = data_path + '/derivatives/freesurfer/subjects'
+    task = 'somato'
+    raw_fname = os.path.join(data_path, f'sub-{subject}', 'meg',
+                             f'sub-{subject}_task-{task}_meg.fif')
+    fwd_fname = os.path.join(data_path, 'derivatives', f'sub-{subject}',
+                             f'sub-{subject}_task-{task}-fwd.fif')
+    condition = 'Unknown'
+
+    # Read evoked
+    raw = mne.io.read_raw_fif(raw_fname)
+    events = mne.find_events(raw, stim_channel='STI 014')
+    reject = dict(grad=4000e-13, eog=350e-6)
+    picks = mne.pick_types(raw.info, meg=True, eeg=True, eog=True)
+
+    event_id, tmin, tmax = 1, -.2, .25
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, picks=picks,
+                        reject=reject, preload=True)
+    evoked = epochs.average()
+    evoked = evoked.pick_types('grad')
+
+    # Compute noise covariance matrix
+    noise_cov = mne.compute_covariance(epochs, rank='info', tmax=0.)
+
+    # Selecting relevant time window: focusing on early signal
+    evoked.plot()
+    t_min, t_max = 0.03, 0.05
+    t_step = 1.0 / 300
+
+    # We must reduce the whitener since data were preprocessed for removal
+    # of environmental noise leading to an effective number of 64 samples.
+    pca = True
```

Comment:
did you try passing rank='info' in prepare_gain?

Reply:
Yes and it is not reducing the whitener as desired: Solution are spurious for somato. 

'rank = None' works well.